scripts/helpers/eosio.sh

```diff
@@ -0,0 +1,551 @@
+# Checks for Arch and OS + Support for tests setting them manually
+## Necessary for linux exclusion while running bats tests/bash-bats/*.sh
+[[ -z "${ARCH}" ]] && export ARCH=$( uname )
+if [[ -z "${NAME}" ]]; then
+    if [[ $ARCH == "Linux" ]]; then
+        [[ ! -e /etc/os-release ]] && echo "${COLOR_RED} - /etc/os-release not found! It seems you're attempting to use an unsupported Linux distribution.${COLOR_NC}" && exit 1
+        # Obtain OS NAME, and VERSION
+        . /etc/os-release
+    elif [[ $ARCH == "Darwin" ]]; then export NAME=$(sw_vers -productName)
+    else echo " ${COLOR_RED}- EOSIO is not supported for your Architecture!${COLOR_NC}" && exit 1
+    fi
+fi
+
+# Setup yum and apt variables
+if [[ $NAME =~ "Amazon Linux" ]] || [[ $NAME == "CentOS Linux" ]]; then
+    if ! YUM=$( command -v yum 2>/dev/null ); then echo "${COLOR_RED}YUM must be installed to compile EOS.IO${COLOR_NC}" && exit 1; fi
+elif [[ $NAME == "Ubuntu" ]]; then
+    if ! APTGET=$( command -v apt-get 2>/dev/null ); then echo "${COLOR_RED}APT-GET must be installed to compile EOS.IO${COLOR_NC}" && exit 1; fi
+fi
+
+# Obtain dependency versions; Must come first in the script
+. ./scripts/.environment
+# Load general helpers
+. ./scripts/helpers/general.sh
+
+function setup() {
+    if $VERBOSE; then
+        echo "VERBOSE: ${VERBOSE}"
+        echo "DRYRUN: ${DRYRUN}"
+        echo "TEMP_DIR: ${TEMP_DIR}"
+        echo "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}"
+        echo "CORE_SYMBOL_NAME: ${CORE_SYMBOL_NAME}"
+        echo "BOOST_LOCATION: ${BOOST_LOCATION}"
+        echo "INSTALL_LOCATION: ${INSTALL_LOCATION}"
+        echo "BUILD_DIR: ${BUILD_DIR}"
+        echo "EOSIO_INSTALL_DIR: ${EOSIO_INSTALL_DIR}"
+        echo "NONINTERACTIVE: ${NONINTERACTIVE}"
+        echo "PROCEED: ${PROCEED}"
+        echo "ENABLE_COVERAGE_TESTING: ${ENABLE_COVERAGE_TESTING}"
+        echo "ENABLE_DOXYGEN: ${ENABLE_DOXYGEN}"
+        echo "ENABLE_MONGO: ${ENABLE_MONGO}"
+        echo "INSTALL_MONGO: ${INSTALL_MONGO}"
+        echo "SUDO_LOCATION: ${SUDO_LOCATION}"
+        echo "PIN_COMPILER: ${PIN_COMPILER}"
+    fi
+    [[ -d $BUILD_DIR ]] && execute rm -rf $BUILD_DIR # cleanup old build directory
+    # Be absolutely sure TEMP_DIR is set (we could do damage here)
+    ## IF user set, they can be responsible for creating and then removing stale data
+    if [[ $TEMP_DIR =~ "/tmp" ]]; then
```

Comment:
This is a protective measure I need for the bats test scripts. It's also protection for users with non-standard/default $TEMP_DIR to prevent them from deleting everything in the directory (they might be using it for other things). It's harmless as far as I can tell.